tests/reftests/repository-update.test

```diff
@@ -0,0 +1,181 @@
+N0REP0
```

Comment:
These tests should be added all at once in the first commit of the PR, ran to unsure the initial state is correct, then ran again for each new commits to show new behaviours

Reply:
we already have a reftest called `update.test` i think it should probably go there instead